sdk/samples/iot/sample_pnp_thermostat_component.h

```diff
@@ -9,6 +9,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+#include <azure/core/az_json.h>
```

Comment:
How did you detect that this needed to be added?

Reply:
It does not compile, the type is unknown. You can see below, it is using json types, but never includes a json header.

Reply:
In that case, how did it pass CI before and get merged to master?

https://github.com/Azure/azure-sdk-for-c/blob/f3a74661625adcf20c1df10ec0cb5e59247f9a9e/sdk/samples/iot/sample_pnp_thermostat_component.h#L4-L16